### PR TITLE
feat: Add calculation_id_by_grid_area_code and energy_supplier_id parameter to settlement report job

### PR DESCRIPTION
**Pull request by JesperGomes**
# Description

## Pull-request quality

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [x] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/databricks/calculation_engine/package/settlement_report_job/settlement_report_args.py [hunk 1]
```diff
@@ -24,6 +24,8 @@ class SettlementReportArgs:
     period_start: datetime
     period_end: datetime
     calculation_type: CalculationType
+    calculation_id_by_grid_area: dict[str, str]
```
Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/settlement_report_job/settlement_report_args.py:27]: What's this? Code doc?

Comment (BjarkeMeier) [inline source/databricks/calculation_engine/package/settlement_report_job/settlement_report_args.py:27]:
```suggestion
    calculation_id_by_grid_area: dict[str, uuid]
```

Comment (JesperGomes) [reply source/databricks/calculation_engine/package/settlement_report_job/settlement_report_args.py:27]: not sure what you mean. It is a parameter to be used in the settlement report generation

Comment (BjarkeMeier) [reply source/databricks/calculation_engine/package/settlement_report_job/settlement_report_args.py:27]: I had to look through the PR to guess its purpose. So a brief docstring about the content and purpose would be useful.

Comment (JesperGomes) [reply source/databricks/calculation_engine/package/settlement_report_job/settlement_report_args.py:27]: Is this worth doing considering  that the data is stored as a string and therefore must be converted back to string in the pyspark code?

Comment (BjarkeMeier) [reply source/databricks/calculation_engine/package/settlement_report_job/settlement_report_args.py:27]: It enhances type safety = reducing the risk of code errors.

Comment (JesperGomes) [reply source/databricks/calculation_engine/package/settlement_report_job/settlement_report_args.py]: I have changed it to uuid

**Review by BjarkeMeier — APPROVED** [review]

**Pull request merged**